File: redsheet-portal/src/app/database-models/project-game-detail.ts
import { required, maxLength, range, nested } from '@rx/annotations';
import { ProjectModule, Game, vProjectGameDetailRecord } from './'
export class ProjectGameDetail {
    constructor(projectGameDetail?: ProjectGameDetail | vProjectGameDetailRecord) {
        let properties = ["notes", "projectGameDetailId", "currentGameId", "projectModuleId", "theirCurrentGameId", "theirEventGameId", "games",];
        for (let property of properties)
            if (projectGameDetail && projectGameDetail[property])
                this[property] = projectGameDetail[property];
    }

    @required()
    @maxLength(200)
    notes: string = undefined;

    projectGameDetailId: number = 0;

    @range(0, 2147483647)
    currentGameId: number = undefined;

    @range(0, 2147483647)
    projectModuleId: number = undefined;
    projectModule: ProjectModule;

    @range(0, 2147483647)
    theirCurrentGameId: number = undefined;

    @range(0, 2147483647)
    theirEventGameId: number = undefined;
    @nested(Game)
    games: Game[];



}

File: RedSheet.DbEntities/Models/ProjectGameDetail.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("ProjectGameDetails", Schema = "dbo")]
	[Serializable]
	public partial class ProjectGameDetail  
	{
 
	[Required] 
	[MaxLength(500)]
	public string Notes { get; set; } 
 
	[DatabaseGenerated(DatabaseGeneratedOption.Identity)] 
	[Key]
	public int ProjectGameDetailId { get; set; } 
	///<summary>
    ///{{GameType}}
    ///</summary> 
	[RelationshipTableAttribue("ApplicationObjects","dbo")]
	public GameType CurrentGameId { get; set; } 
 
	[RelationshipTableAttribue("ProjectModules","dbo")]
	//Relationships
	public int ProjectModuleId { get; set; } 
 
	[ForeignKey("ProjectModuleId")]
	public ProjectModule ProjectModule { get; set; } 
	///<summary>
    ///{{GameType}}
    ///</summary> 
	[RelationshipTableAttribue("ApplicationObjects","dbo")]
	public GameType TheirCurrentGameId { get; set; } 
	///<summary>
    ///{{GameType}}
    ///</summary> 
	[RelationshipTableAttribue("ApplicationObjects","dbo")]
	public GameType TheirEventGameId { get; set; } 
	[InverseProperty("ProjectGameDetail")]
	public ICollection<Game> Games { get; set; } 


	public ProjectGameDetail()
	{
	this.Games = new HashSet<Game>();

    }

    }
}

